### DISCUSSION TITLE
Arduino-pico and external EEPROM

### DISCUSSION THREAD

**Discussion by fletchjp** [earlephilhower/arduino-pico#1026] [General]
I am attempting to connect an external EEPROM AT24C256 to a Raspberry Pi Pico using I2C via pins GP6 and GP7.

A scan for the I2C device finds 0x50 and 0x58 as on other hardware (e.g. Arduino DUE).

Write apparently succeeds and read fails.

I am using the latest Arduino-Pico 2.6.4. and Arduino IDE 1.8.19.  I am using Wire1 and pins GP6 and GP7.

Do you have any relevant examples?

Thanks

John

**Comment by earlephilhower** [comment]
Sorry, no.  As this is a general programming Q, I'm moving to a discussion for followup.  I'd suggest looking at some of the existing Arduino libraries for talking to your chip for starters...

**Comment by fletchjp** [comment]
Thank you.  I now have something working, I have found out what I was doing wrong.  John

**Reply by ian-jeffery** [reply]
Can you let us know what the problem was and how you resolved it - it really REALLY helps other people who have similar problems who might have found this using google

**Reply by fletchjp** [reply]
Ian   My problems arose mainly from the fact that I was modifying code that runs on a different processor, I had not appreciated the need to set the SDA and SCL pins to suitable values for working with Wire1, as well as the I2C address ()x50). Once I had these sorted out it started to work. John

**Reply by ian-jeffery** [reply]
Hi John,
I too am also in a similar position. I can scan for the eeprom chip, and it sees it, but it cant read from it? I have set the pins as you have done ( mine are 8 and 9 ) but still no joy. Can you share your code with me please to help me ?
Ian.

**Reply by ixin91** [reply]
Arduino IDE 1.8.16.
Arduino-Pico 2.6.5.
EEPROM AT24C256 , also found two addresses 0x50 and 0x58
I need GP16 and GP17, so I only test with them.
Found on the Internet....
[PicoPi_AT24C256_OK.txt](https://github.com/earlephilhower/arduino-pico/files/10298429/PicoPi_AT24C256_OK.txt)

**Reply by ixin91** [reply]
With Pico inputs and delay for Serial....
[PicoPi_AT24C256_OK.txt](https://github.com/earlephilhower/arduino-pico/files/10298444/PicoPi_AT24C256_OK.txt)
